Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -52,6 +52,41 @@ public class GraphTraverser: GraphTraversing {
         allTargets(excludingExternalTargets: true)
     }
 
+    public func filterIncludedTargets<GraphTargets: Collection>(
+        basedOn targets: GraphTargets,
```

Comment:
We could - right now the function may be called with an `Array` or `Set` in the current implementation, so making it generic instead of converting to `Set` made sense to me. Down to change if that's not usually done in the codebase.

qq as well: I think work is still ongoing with Xcode 14.2 & Swift 5.7, but once available would it work to just use `some Collection<GraphTarget>`?

Reply:
yep, `some Collection<GraphTarget>` would be ideal, let's keep it as it is for now 🙏